docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,446 @@
+.. _building_the_project_with_dpcpp:
+
+Building the Project with DPC++
+===============================
+
+This page describes building the oneMKL interface library with either the Intel(R) oneAPI DPC++ Compiler or open-source oneAPI DPC++ Compiler.
+For guidance on building the project with AdaptiveCpp, see :ref:`building_the_project_with_adaptivecpp`.
+
+.. _build_setup_with_dpcpp:
+
+Build Setup
+###########
+
+#. 
+   Install Intel(R) oneAPI DPC++ Compiler (select the variant as described in
+   :ref:`Selecting a Compiler<selecting_a_compiler>`).
+
+#. 
+   Clone this project to ``<path to onemkl>``\ , where ``<path to onemkl>``
+   is the root directory of this repository.
+
+#. 
+   Build and install all `required dependencies <https://github.com/oneapi-src/oneMKL?tab=readme-ov-file#software-requirements>`_. 
+
+.. _build_introduction_with_dpcpp:
+
+Introduction
+############
+
+In most cases, building oneMKL is as simple as setting the compiler and selecting the desired backends to build with.
+
+On Linux (see `Building for Windows`_ for building on Windows):
+
+  .. code-block:: bash
+
+     # Inside <path to onemkl>
+     mkdir build && cd build
+     cmake .. -DCMAKE_CXX_COMPILER=$CXX_COMPILER                        \ # Should be icpx or clang++
+              -DCMAKE_C_COMPILER=$C_COMPILER                            \ # Should be icx or clang
+              -DENABLE_MKLGPU_BACKEND=OFF                               \ # Optional: The MKLCPU backend is ON by default.
+              -DENABLE_MKLGPU_BACKEND=OFF                               \ # Optional: The MKLGPU backend is ON by default.
+              -DENABLE_<BACKEND_NAME>_BACKEND=ON                        \ # Enable any other backend(s) (optional)
+              -DENABLE_<BACKEND_NAME_2>_BACKEND=ON                      \ # Multiple backends can be enabled at once.
+              -DBUILD_FUNCTIONAL_TESTS=OFF                              \ # See page *Building and Running Tests* for more on building tests. ON by default.
+              -DBUILD_EXAMPLES=OFF                                        # Optional: On by default.
+     cmake --build .
+     cmake --install . --prefix <path_to_install_dir>                    # required to have full package structure
```

Comment:
probably need to unindent this block

Reply:
Done for this and other blocks in [43e0a04](https://github.com/oneapi-src/oneMKL/pull/510/commits/43e0a04c6a415beabc04d8058f7884073f62c46f).